go/mysql/conn.go

```diff
@@ -917,7 +917,13 @@ func (c *Conn) handleNextCommand(handler Handler) error {
 		}
 
 		if len(queries) != 1 {
-			return fmt.Errorf("can not prepare multiple statements")
+			err := fmt.Errorf("can not prepare multiple statements")
+			if werr := c.writeErrorPacketFromError(err); werr != nil {
+				// If we can't even write the error, we're done.
+				log.Errorf("Error writing query error to %s: %v", c, werr)
+				return werr
+			}
+			return err
```

Comment:
Looks like any error returned by this function kills the server. So you should return nil here